oneflux_steps/ustar_cp_refactor_wip/launch.m

```diff
@@ -7,369 +7,162 @@
 %	load and assign data
 
 function exitcode = launch(input_folder, output_folder)
-exitcode = 0;
-warning off;
+    exitcode = 0;
+    warning off;
 
-% check input path
-if 0 == exist('input_folder')
-    input_folder = [pwd '/'];
-elseif length(input_folder) < 2
-    if input_folder(2) ~= ':'
-        input_folder = [pwd '/' input_folder];
-    end
-end
-if input_folder(length(input_folder)) ~= '\' && input_folder(length(input_folder)) ~= '/'
-    input_folder = [input_folder '/'];
-end
+    [input_folder, output_folder] = checkPath(input_folder, output_folder);
 
-% check output path
-if 0 == exist('output_folder')
-    output_folder = [pwd '/'];
-elseif length(output_folder) < 2 
-    if output_folder(2) ~= ':'
-        output_folder = [pwd '/' output_folder];  
-    end
-end
-if output_folder(length(output_folder)) ~= '\' && output_folder(length(output_folder)) ~= '/'
-    output_folder = [output_folder '/'];
-end
-mkdir(output_folder);
+    %by alessio
+    USTAR_INDEX = 1;
+    NEE_INDEX = 2;
+    TA_INDEX = 3;
+    PPFD_INDEX = 4;
+    RG_INDEX = 5;
+    input_columns_names = {'USTAR' 'NEE' 'TA' 'PPFD_IN' 'SW_IN' };
+
+    %by alessio
+    fprintf('\n\nUstar Threshold Computation by Alan Barr\n\ninput in %s\noutput in %s\n\n', input_folder, output_folder);
+    d = dir([input_folder,'*_qca_ustar_*.csv']);
+    error_str = {};
 
-%by alessio
-USTAR_INDEX = 1;
-NEE_INDEX = 2;
-TA_INDEX = 3;
-PPFD_INDEX = 4;
-RG_INDEX = 5;
-input_columns_names = {'USTAR' 'NEE' 'TA' 'PPFD_IN' 'SW_IN' };
+    % by alessio
+    fprintf('%d files founded.\n\n', numel(d));
+
+    for n = 1:numel(d)
+        % by alessio
+        fprintf('processing n.%02d, %s...', n, d(n).name);
+        metadata = struct();
+        metadata.siteFile = split(d(n).name, '.');
+        metadata.siteFile = metadata.siteFile{1};

```

Comment:
Bit unclear what this is for (`metadata` is not used after this point). Is this for our testing framework purposes? Could you add a comment if so.

Reply:
Added these when I was logging the input and output but was supposed to delete it.